C7Engine/AI/Pathing/DijkstrasLandAlgorithm.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using C7GameData;
+
+namespace C7Engine.Pathing
+{
+	/**
+	 * Uses Dijkstra's Algorithm to find a path between two tiles.
+	 * Advantages: Finds the shortest path accounting for tile movement cost
+	 * Disadvantages: Expensive, will exhaustively search every tile on the continent
+	 *
+	 * Notes: - Edge weight is defined by the movement cost of the 2nd node:
+	 *          w(u, v) = movement_cost(v)
+	 */
+	public class DijkstrasLandAlgorithm : PathingAlgorithm
+	{
+		//N.B. This should really be static, but we can't put a static method on interfaces, so it isn't.
+		public TilePath PathFrom(Tile start, Tile destination)
+		{
+			// shortest distance from start to each tile on the continent
+			Dictionary<Tile, int> dist = new Dictionary<Tile, int>();
+			Dictionary<Tile, Tile> predecessors = new Dictionary<Tile, Tile>();
+			HashSet<Tile> visited = new HashSet<Tile>();
+
+			dist[start] = 0;
+
+			bool updateShortestDistance(Tile tile, int distance) {
```

Comment:
I try not to notpick style in code reviews (especially before we had the .editorconfig)... but why not make this a private helper method instead of an line one?  It would interrupt the flow of reading the method significantly less as a helper method than being inline.

Reply:
Ah, perhaps because it modifies the dist Dictionary?  I'd still favor kicking it out into a separate method.

Reply:
The main reason was because I didn't have to type out as many parameters, and I have no objection to changing it to a helper if it improves readability. The other reason is because since it modifies `dist`, I prefer a nested method since `dist` can only refer to the `dist` in the outer scope and my IDE is aware of this, whereas with a helper method you have to figure out that you're passing in a mutable/output parameter which is actual `dist` in the main dijsktra's function

Reply:
Hmm... you're right, dist would be a mutable parameter (although I believe it wouldn't technically be a C# `out` parameter; you could modify the collection with it being passed as a "regular" parameter).

I think it comes down to style and background.  I've mostly worked in Java/Kotlin/TypeScript, and given the length of these inline functions, they'd usually be kicked out.  A lambda that is a couple lines, or a short anonymous inner function, might stick around, but if the inner function rivals the length of the parent one, it should probably be its own named function; I'd expect my colleagues to request I move one of this length.  But it's also not a hard-and-fast boundary, and I'm sure I have some anonymous inner functions of humungous length in my editor from back in the day.

I would ask what your favorite IDE is here, but I think instead I'll create a poll on GitHub Discussions to try that out.